src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_NUMBER = 10;
+    private Pair<K, V>[] pairs;
+
+    public StorageImpl() {
+        this.pairs = new Pair[MAX_NUMBER];
+    }
+
     @Override
-    public void put(K key, V value) {
+    public void put(K key,V value) {
+        int filledSize = size();
+        if (filledSize == 0) {
+            pairs[filledSize] = new Pair(key, value);
+        }
+        if (filledSize <= MAX_NUMBER) {
+            if (get(key) == null) {
+                for (int i = 0; i < pairs.length; i++) {
+                    if (pairs[i] == null) {
+                        pairs[i] = new Pair(key, value);
+                        break;
+                    }
+                }
+            }
+            if (get(key) != null) {
+                for (int i = 0; i < filledSize; i++) {
+                    if ((pairs[i].myKey == null && key == null)
+                            || ((pairs[i].myKey != null && key != null)
+                                    && pairs[i].myKey.equals(key))) {
+                        pairs[i].myValue = value;
+                    }
+                }
+            }
+        }
+
     }
 
     @Override
     public V get(K key) {
+        int length = size();
+        for (int i = 0; i < length; i++) {
+            if ((pairs[i].myKey == null && key == null)
+                    || ((pairs[i].myKey != null && key != null)
+                           && pairs[i].myKey.equals(key))) {
+                return (V) pairs[i].myValue;
+            }
+        }
```

Comment:
try to simplify this method
```suggestion
        for (int i = 0; i < size; i++) {
            if (compare(keys[i], key)) {
                return values[i];
            }
        }
```
and add the next method:
```
    private boolean compare(K key1, K key2) {
        return key1 == key2 || key1 != null && key1.equals(key2);
    }
```